scikeras/utils/transformers.py

```diff
@@ -0,0 +1,197 @@
+            raise ValueError(
+                f"Unknown label type: {target_type}."
+                "\n\nTo implement support, subclass KerasClassifier and override"
+                " `target_transformer` with a transformer that supports this"
+                " label type."
+            )
+        self.final_encoder_ = encoders[target_type].fit(y)
+
+        if (
+            target_type == "multilabel-indicator"
+            and y.min() == 0
+            and (y.sum(axis=1) == 1).all()
+        ):
+            target_type = "multiclass-onehot"
+
+        self.n_outputs_ = 1
+        self.model_n_outputs_ = 1
+        self.y_dtype_ = y.dtype
+        self._target_type = target_type
+
+        if target_type in ("binary", "multiclass"):
+            self.classes_ = self.final_encoder_[1].categories_[0]
+            self.n_classes_ = self.classes_.size
+        elif target_type in ("multiclass-onehot", "multilabel-indicator"):
+            self.classes_ = np.arange(0, y.shape[1])
+            self.n_classes_ = y.shape[1]
+        elif target_type == "multiclass-multioutput":
+            self.classes_ = None
+            self.n_classes_ = None
+
+        return self
+
+    def transform(self, X: np.ndarray) -> np.ndarray:
+        y = X  # rename for clarity, the input is always expected to be a target `y`
+        # no need to validate n_outputs_ or model_n_outputs_, those are hardcoded
+        # self.classes_ and self.n_classes_ are validated by the transformers themselves
+        return self.final_encoder_.transform(y)
+
+    def inverse_transform(
+        self, X: np.ndarray, return_proba: bool = False
+    ) -> np.ndarray:
+        y = X  # rename for clarity, the input is always expected to be a target `y`
+        if self._target_type == "binary":
+            # array([0.9, 0.1], [.2, .8]) -> array(['yes', 'no'])
+            if y.ndim == 1 or (y.shape[1] == 1 and self.n_classes_ == 2):
+                # result from a single sigmoid output
+                # reformat so that we have 2 columns
+                y = np.column_stack([1 - y, y])
+            y_ = np.argmax(y, axis=1).reshape(-1, 1)
+            class_predictions = self.final_encoder_.inverse_transform(y_)
+        elif self._target_type == "multiclass":
+            # array([0.8, 0.1, 0.1], [.1, .8, .1]) ->
+            # array(['apple', 'orange'])
+            idx = np.argmax(y, axis=-1)
+            if not is_categorical_crossentropy(self.loss):
+                y_ = idx.reshape(-1, 1)
+            else:
+                y_ = np.zeros(y.shape, dtype=int)
+                y_[np.arange(y.shape[0]), idx] = 1
+            class_predictions = self.final_encoder_.inverse_transform(y_)
+        elif self._target_type == "multiclass-onehot":
+            # array([.8, .1, .1], [.1, .8, .1]) ->
+            # array([[1, 0, 0], [0, 1, 0]])
+            idx = np.argmax(y, axis=-1)
+            y_ = np.zeros(y.shape, dtype=int)
+            y_[np.arange(y.shape[0]), idx] = 1
```

Comment:
Style nit:

``` diff
-             y_[np.arange(y.shape[0]), idx] = 1
+             y_[:, idx] = 1
```

Reply:
Another style nit:

``` python
class_predictions = np.zeros(...)
class_predictions[:, idx] = 1
```

Reply:
The first part is a no brainer and a total brain fart on my end.

The second style nit is just about renaming the variable right? I think I wanted to keep probability (including their transformations) separate from the class predictions by not using the `class_predictions` variable for intermediately manipulations of the probabilities. Am I understanding your suggestion correctly, and do you still want to go ahead with it?

Reply:
> The second style nit is just about renaming the variable right? 

Correct.

> I think I wanted to keep probability (including their transformations) separate from the class predictions by not using the class_predictions variable for intermediately manipulations of the probabilities.

I added this suggestion because I got confused between `y_` and `y`. I don't think renaming is a huge issue because `class_predictions` never changes type.

> do you still want to go ahead with it?

That's my suggestion.

Reply:
Implemented